src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -723,6 +758,52 @@ void FancyZones::MoveSizeUpdateInternal(HMONITOR monitor, POINT const& ptScreen,
     }
 }
 
+void FancyZones::HandleVirtualDesktopUpdates(HANDLE fancyZonesDestroyedEvent) noexcept
+{
+    HANDLE regKeyEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+    HANDLE events[2] = { regKeyEvent, fancyZonesDestroyedEvent };
+    while (1) {
+        if (RegNotifyChangeKeyValue(HKEY_CURRENT_USER, TRUE, REG_NOTIFY_CHANGE_LAST_SET, regKeyEvent, TRUE) != ERROR_SUCCESS) {
+            return;
+        }
+        if (WaitForMultipleObjects(2, events, FALSE, INFINITE) != (WAIT_OBJECT_0 + 0)) {
+            // if fancyZonesDestroyedEvent is signalized or WaitForMultipleObjects failed, terminate thread execution
+            return;
+        }
+        DWORD bufferCapacity;
+        const WCHAR* key = L"VirtualDesktopIDs";
+        // request regkey binary buffer capacity only
+        if (RegQueryValueExW(m_virtualDesktopsRegKey, key, 0, nullptr, NULL, &bufferCapacity) != ERROR_SUCCESS) {
+            return;
+        }
+        BYTE* buffer = (BYTE*)malloc(bufferCapacity);
+        // request regkey binary content
+        if (RegQueryValueExW(m_virtualDesktopsRegKey, key, 0, nullptr, buffer, &bufferCapacity) != ERROR_SUCCESS) {
+            free(buffer);
+            return;
+        }
+        const int guidSize = sizeof(GUID);
+        std::unordered_map<GUID, bool> temp;
+        for (int i = 0; i < bufferCapacity; i += guidSize) {
+            GUID guid;
+            memcpy(&guid, buffer + i, guidSize);

```

Comment:
no need to memcpy stuff, we could just use reinterpret_cast and do temp[*guid]

Reply:
`GUID* guid = reinterpret_cast<GUID*>(buffer + i)` and `temp[*guid] = true;` ?